### PR TITLE
Add optional seed

### PR DESCRIPTION
**Pull request by mark-todd**

### CODE REVIEW

File: epioncho_ibm/advance/blackfly.py [hunk 1]
```diff
@@ -237,5 +239,5 @@ def calc_new_worms_from_blackfly(
     )
     if debug:
         assert not np.any(new_rate > 10**10)
```
Comment (mkcinu) [inline epioncho_ibm/advance/blackfly.py:241]: Haha, lol - that's a big number!

File: epioncho_ibm/state/derived_params.py [hunk 2]
```diff
@@ -78,18 +78,37 @@ def __init__(self, params: Params) -> None:
...
-        self.worm_sex_ratio_generator = Generator(SFC64(), params.worms.sex_ratio)
-        self.worm_lambda_zero_generator = Generator(
-            SFC64(), params.worms.lambda_zero * params.delta_time
-        )
-        self.worm_omega_generator = Generator(
-            SFC64(), params.worms.omega * params.delta_time
-        )
-
-        self.worm_mortality_generator = Generator(SFC64(), self.worm_mortality_rate)
+        if seed is None:
```
Comment (mkcinu) [inline epioncho_ibm/state/derived_params.py:81]: Yuck. How about this:
`bit_generators = [SFC64((seed + i + 1) if seed is not None else None for i in range(6))`
and then use `Generator(bit_generators[0])...`, ..., `Generator(bit_generators[5])`
Not very important, but the code repetition isn't nice.

File: epioncho_ibm/state/derived_params.py [hunk 3]
```diff
@@ -25,7 +25,7 @@ class DerivedParams:
     worm_lambda_zero_generator: Generator
     worm_omega_generator: Generator
 
-    def __init__(self, params: Params) -> None:
+    def __init__(self, params: Params, seed: Optional[int]) -> None:
```
Comment (mkcinu) [inline epioncho_ibm/state/derived_params.py:28]: `seed` is a part of `Params`. Why pass it separately?

File: epioncho_ibm/advance/blackfly.py (hunk 1 above)
Comment (mark-todd) [reply epioncho_ibm/advance/blackfly.py:241]: Yeah haha - this is the point where the calculation breaks down I think - determined it as some threshold a while back

File: epioncho_ibm/state/derived_params.py (hunk 2 above)
Comment (mark-todd) [reply epioncho_ibm/state/derived_params.py:81]: Yeah I agree

File: epioncho_ibm/state/derived_params.py (hunk 3 above)
Comment (mark-todd) [reply epioncho_ibm/state/derived_params.py:28]: Yep

File: epioncho_ibm/advance/blackfly.py [hunk 4]
```diff
@@ -216,6 +217,7 @@ def calc_new_worms_from_blackfly(
     n_people: int,
     current_worms: WormGroup,
     debug: bool,
+    numpy_bit_gen: Generator,
```
Comment (mkcinu) [inline epioncho_ibm/advance/blackfly.py:220]: docstring

**Pull request merged**